src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs

```diff
@@ -46,28 +46,45 @@
             "+"]])]))})
 
 
-(react/defc FilterButtons
-  (let [Button
-        (react/create-class
-          {:render
-           (fn [{:keys [props]}]
-             [:div {:style {:float "left" :textAlign "center"
-                            :backgroundColor (if (:active? props)
-                                               (:button-blue style/colors)
-                                               (:background-gray style/colors))
-                            :color (when (:active? props) "white")
-                            :marginLeft "1em" :padding "1ex" :width "16ex"
-                            :border (str "1px solid " (:line-gray style/colors))
-                            :borderRadius "2em"
-                            :cursor "pointer"}
-                    :onClick (fn [e] ((:onClick props) e))}
-              (:text props)])})]
-    {:render
-     (fn [{:keys [props]}]
-       [:div {:style {:display "inline-block" :marginLeft "-1em"}}
-        (map (fn [button] [Button button])
-          (:buttons props))
-        (common/clear-both)])}))
+(react/defc FilterBar
+  {:get-initial-state
+   (fn [{:keys [props]}]
+     {:selected-index 0
+      :filtered-data (into {} (map-indexed
```

Comment:
This pattern causes pieces of the table rows to be stored in the state of a few different components. If I say, "can you make this component work more like Paginator," would you know what I mean?